src/mlpack/bindings/R/mlpack/inst/include/mlpack.h.in

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file inst/include/mlpack.h
+ * @author Yashwant Singh Parihar
+ *
+ * Include all of the base components required to work mlpack bindings.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_R_RCPP_MLPACK_H
+#define MLPACK_BINDINGS_R_RCPP_MLPACK_H
+
+#include <Rcpp.h>
+
+// To suppress Found ‘__assert_fail’, possibly from ‘assert’ (C).
+#define BOOST_DISABLE_ASSERTS
+
+// Rcpp has its own stream object which cooperates more nicely with R's i/o
+// And as of armadillo and mlpack, we can use this stream object as well.
+#if !defined(ARMA_COUT_STREAM)
+  #define ARMA_COUT_STREAM Rcpp::Rcout
+#endif
+#if !defined(ARMA_CERR_STREAM)
+  #define ARMA_CERR_STREAM Rcpp::Rcerr
+#endif
+#if !defined(MLPACK_COUT_STREAM)
+  #define MLPACK_COUT_STREAM Rcpp::Rcout
+#endif
+#if !defined(MLPACK_CERR_STREAM)
+  #define MLPACK_CERR_STREAM Rcpp::Rcerr
+#endif
+
+// This define makes the R RNG have precedent over the C++11-based
+// RNG provided by Armadillo.
+#if !defined(ARMA_RNG_ALT)
+  #define ARMA_RNG_ALT         RcppArmadillo/Alt_R_RNG.h
+#endif
+
+// To suppress warnings related to core/util/arma_util.hpp.
+#define MLPACK_CORE_UTIL_ARMA_CONFIG_HPP
+
+@R_SRC@
+
+#include <mlpack/core.hpp>
+
+// These are all the boost files, we need to compile R bindings for mlpack that
+// are not a part of mlpack itself.
+#include <boost/serialization/archive_exception.cpp>
+#include <boost/serialization/basic_archive.cpp>
+#include <boost/serialization/basic_oarchive.cpp>
+#include <boost/serialization/basic_oserializer.cpp>
+#include <boost/serialization/basic_iarchive.cpp>
+#include <boost/serialization/basic_iserializer.cpp>
+#include <boost/serialization/basic_pointer_iserializer.cpp>
+#include <boost/serialization/basic_pointer_oserializer.cpp>
+#include <boost/serialization/basic_text_oprimitive.cpp>
+#include <boost/serialization/binary_iarchive.cpp>
+#include <boost/serialization/binary_oarchive.cpp>
+#include <boost/serialization/extended_type_info_typeid.cpp>
+#include <boost/serialization/extended_type_info.cpp>
+#include <boost/serialization/basic_serializer_map.cpp>
+#include <boost/serialization/void_cast.cpp>
+#include <boost/serialization/utf8_codecvt_facet.cpp>
```

Comment:
I know you did a bunch of work to make this function right, but hopefully soon we can remove it and replace with cereal (#2458). :)

Reply:
Here I have a question, should we provide a copy of `cereal` with package, if I am not mistaken -- [Rcereal](https://github.com/wush978/Rcereal) is not maintain anymore.

Reply:
BTW #2458 Is giving some great results with the R-Bindings. I have tried to apply cereal serialization on R-bindings [here](https://github.com/Yashwants19/RcppMLPACK/pull/13) with some edits. It got some awesome results, by the help of cereal serialization we were successful in reducing the size of `mlpack.so` file:
* Linux (without strip) from 365.9Mb to 322.7Mb
* Linux (with strip, but locally) from 42.7Mb to 17.0Mb
* macOS from 44.7Mb to 24.3Mb
* Windows from 54.0Mb to 30.9Mb

For more information you can compare both checks here: 
  1) [Github action](https://github.com/Yashwants19/RcppMLPACK/runs/898506733?check_suite_focus=true) with boost serialization.
  2) [Github action](https://github.com/Yashwants19/RcppMLPACK/runs/997991588?check_suite_focus=true) with cereal serialization.

Reply:
Awesome!  Let's handle the cereal discussion once #2458 is merged, but I think maybe the best option will be to simply package it with mlpack, unless we want to also resurrect and maintain Rcereal (not sure what's easier; I assume just bundling a cereal version would be easiest).